src/controllers/user.controller.js

```diff
@@ -136,10 +136,14 @@ export async function getUsers(req, res) {
 }
 
 export async function getAuthority(req, res) {
-    const accessToken =
-        req.headers["authorization"].split(" ")[1] ||
-        req.headers["Authorization"].split(" ")[1];
-    const decoded = jwt.verify(accessToken, process.env.JWT_SECRET);
-    const authority = decoded.authority;
-    res.status(200).json({ authority: authority });
+    if (!req.headers["authorization"] || !req.headers["Authorization"]) {
+        return res.status(401).json({ message: "No token provided" });
+    } else {
+        const accessToken =
+            req.headers["authorization"].split(" ")[1] ||
+            req.headers["Authorization"].split(" ")[1];
+        const decoded = jwt.verify(accessToken, process.env.JWT_SECRET);
+        const authority = decoded.authority;
+        res.status(200).json({ authority: authority });
+    }
```

Comment:
```suggestion
    const auth = req.headers?.["authorization"] ?? req.headers?.["Authorization"];
    const accessToken = auth?.split(" ")?.[1];
    
    try {
        if (!auth || !accessToken) {
            throw new Error();
        } else {
            const decoded = jwt.verify(accessToken, process.env.JWT_SECRET);
            const authority = decoded.authority;
            res.status(200).json({ authority: authority });
        }
    } catch {
        return res.status(401).json({ message: "No token provided or it's malformed" });
    }
```
Untested, so copy and paste first to check it works properly please. In my opinion, this solution is more concise:
* No double access to the headers (your solution did it 4 times, in if check and in else).
* Uses optional chaining to reduce verbosity
* Catches all kind of errors by wrapping everything in a try/catch and throwing when the token is missing. This way we also handle errors in ``jwt.verify``, which could happen if the auth header is also invalid. Previously you were assuming that, beside formatting, everything else would be OK, but that might not be the case and we should never trust the client.

Reply:
You are right, your solution efficiently solves the problem.